src/orient/rational-calc.orient

```diff
@@ -26,22 +26,27 @@ ProofOfSpacetime:
 
     // TODO: add MerkleTreeCaching for ColumnR
     MerkleTreeCaching:
+      Storage:
+        post_mtree_cached = post_mtree_cached_hashnodes * node_size
+        post_mtree_cached_gb = post_mtree_cached / (1024 * 1024 * 1024)
+        post_mtree_overhead = post_mtree_cached / sector_size
+
       // TODO: make sure this is correct
-      post_cached_regen_hashes = expt(2, (log2(nodes) - post_cached_layers))
-      // TODO: check if post_merkle_region_hashes shoulld be *2
-      post_cached_regen_challenge_time = (post_cached_regen_hashes * 2) * merkle_tree_hash_time
-      post_cached_regen_challenges_time = post_cached_regen_challenge_time * post_challenges
-      post_cached_lookup_challenges_time = post_cached_regen_hashes * rig_storage_latency
-      // log2(nodes) > post_cached_layers
-
-      post_cached_hashes = expt(2, post_cached_layers) + 1
-      post_cached_remaining_layers = commr_tree_depth - post_cached_layers
-      post_cached = post_cached_hashes * node_size
-      post_cached_gb = post_cached / (1024 * 1024 * 1024)
-      post_cached_fraction = post_cached / sector_size
+      Regeneration:
+        post_mtree_cached_hashnodes = expt(2, post_mtree_layers_cached) + 1
+        post_mtree_layers_deleted = commr_tree_depth - post_mtree_layers_cached
+
+        post_mtree_regen_nodes = expt(2, (log2(nodes) - post_mtree_layers_cached))
```

Comment:
Isn't `log2(nodes)` just `commr_tree_depth`?

Let's use that instead. But then the whole exponent just becomes `post_mtree_layers_deleted`.

Let's use that instead. But then we can see that this is off by one. I think what we want is `post_mtree_regen_nodes = expt(2, post_mtree_layers_deleted) - 1`.

Reply:
resolved in latest